.github/workflows/tests.yaml

```diff
@@ -108,12 +108,21 @@ jobs:
           mkdir -p artifacts
           mv tests/integration/wasms/* artifacts
       - name: 'Upload Artifacts'
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: integration-tests
+          name: integration-tests-${{ matrix.canister }}
           path: artifacts/${{ matrix.canister }}.wasm.gz
           retention-days: 10
           if-no-files-found: error
+  merge-artifacts:
+    runs-on: ubuntu-latest
+    needs: [build-wasms, download-canisters]
+    steps:
+      - name: Merge Artifacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: integration-tests
+          pattern: integration-tests-*
   integration-tests:
     name: 'integration-tests:required'
     needs: [build-wasms, download-canisters]
```

Comment:
wouldn't this need to depend on `merge-artifacts` now?

Reply:
indeed, nice catch!